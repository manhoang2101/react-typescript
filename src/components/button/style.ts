import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    appButton: {
      "&:hover": {
        backgroundColor: "#fff",
        borderColor: "#fff",
        boxShadow: "none",
      },
      "&:active": {
        boxShadow: "none",
        backgroundColor: "#0062cc",
        borderColor: "#005cbf",
      },
      "&:focus": {
        boxShadow: "0 0 0 0.2rem rgba(0,123,255,.5)",
      },
    },
  });

export default style;
