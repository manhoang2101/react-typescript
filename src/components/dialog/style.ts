import { Theme, createStyles } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    form: {
      display: "flex",
      flexDirection: "column",
      margin: "auto",
      width: "fit-content",
    },
    formControl: {
      marginTop: theme.spacing(2),
      minWidth: 120,
    },
    formControlLabel: {
      marginTop: theme.spacing(1),
    },
    appBar: {
      position: "relative",
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
    title: {
      marginLeft: 0,
      flex: 1,
    },
    gutters: {
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
  });

export default style;
