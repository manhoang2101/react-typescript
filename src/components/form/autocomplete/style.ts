import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    TextField: {
      paddingRight: 25,
    },

    MuiAutocomplete: {
      input: {
        paddingRight: 25,
      },
    },
    KeyboardArrowDownIcon: {
      paddingLeft: 0,
      borderLeft: `1px solid #ccc`,
    },
  });

export default style;
