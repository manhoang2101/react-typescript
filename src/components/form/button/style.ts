import { createStyles } from "@material-ui/core";

const style = () => createStyles({});

export default style;
