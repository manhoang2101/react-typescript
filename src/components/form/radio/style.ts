import { createStyles, Theme } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    MuiFormHelperText: {
      marginLeft: 0,
    },
  });

export default style;
