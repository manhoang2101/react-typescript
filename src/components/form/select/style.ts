import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    formControl: {
      minWidth: 120,
    },
  });

export default style;
