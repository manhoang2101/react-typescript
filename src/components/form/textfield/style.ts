import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    AppTextField: {
      display: "block",
    },
  });

export default style;
