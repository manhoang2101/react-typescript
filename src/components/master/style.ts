import { createStyles, Theme, fade } from "@material-ui/core";
import { drawerWidth } from "./constants";

const style = (theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
    },
    // header styles
    grow: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      display: "none",
      [theme.breakpoints.up("lg")]: {
        display: "block",
      },
      color: theme.palette.common.white,
    },
    search: {
      position: "relative",
      borderRadius: theme.shape.borderRadius,
      backgroundColor: fade(theme.palette.common.white, 0.15),
      "&:hover": {
        backgroundColor: fade(theme.palette.common.white, 0.25),
      },
      marginRight: theme.spacing(2),
      marginLeft: 0,
      width: "100%",
      [theme.breakpoints.up("sm")]: {
        marginLeft: theme.spacing(3),
        width: "auto",
      },
    },
    searchIcon: {
      width: theme.spacing(7),
      height: "100%",
      position: "absolute",
      pointerEvents: "none",
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
    },
    inputRoot: {
      color: "inherit",
    },
    inputInput: {
      padding: theme.spacing(1, 1, 1, 7),
      transition: theme.transitions.create("width"),
      width: "100%",
      [theme.breakpoints.up("md")]: {
        width: 200,
      },
    },
    sectionDesktop: {
      display: "none",
      [theme.breakpoints.up("md")]: {
        display: "flex",
      },
    },
    sectionMobile: {
      display: "flex",
      [theme.breakpoints.up("md")]: {
        display: "none",
      },
    },
    appBar: {
      transition: theme.transitions.create(["margin", "width"], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    appBarShift: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
      transition: theme.transitions.create(["margin", "width"], {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },

    // !header styles

    hide: {
      display: "none",
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
    },
    drawerPaper: {
      width: drawerWidth,
    },
    drawerHeader: {
      display: "flex",
      alignItems: "center",
      padding: theme.spacing(0, 1),
      // necessary for content to be below app bar
      ...theme.mixins.toolbar,
      justifyContent: "flex-end",
    },
  });
export default style;
