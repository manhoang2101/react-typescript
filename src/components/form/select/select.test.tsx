import { withStyles } from "@material-ui/core";
import style from "./style";
import AppSelect, { ISelectItem } from ".";
import { shallow } from "enzyme";
import React from "react";
import { render } from "@testing-library/react";

describe("<AppSelect />", () => {
  const Component = withStyles(style)(AppSelect);
  test("toMatchSnapshot()", () => {
    const items: ISelectItem[] = [
      { label: "Radio1", value: "Radio1" },
      { label: "Radio2", value: "Radio2" },
      { label: "Radio3", value: "Radio3" },
      { label: "Radio4", value: "Radio4" },
    ];
    const container = shallow(<Component selectItems={items} id="DE" />);
    expect(container).toMatchSnapshot();
  });
  describe("handleChange", () => {
    test("should call handleChange() not multiple and optionSelectAll  and emtySelectOption", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          value={"100"}
        />
      );
      let component = wrapper.dive().dive().instance();
      const event = {
        target: { value: "100" },
      };
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalled();
    });
    test("should call handleChange() not multiple and optionSelectAll  has emtySelectOption", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          emtySelectOption
          value={""}
        />
      );
      let component = wrapper.dive().dive().instance();
      expect(component.props.emtySelectOption).toBe(true);
      const event = {
        target: { value: ["1", "2", "3"] },
      };
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalled();
    });
    test("should call handleChange() has multiple and  not optionSelectAll  has emtySelectOption", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          emtySelectOption
          multiple
          value={["1"]}
          options={{
            emtyOption: {
              label: "-- Choose a option --",
              value: "choose-emty",
            },
            selectAll: {
              label: "Choose All",
              value: "choose-all",
            },
            lables: {
              others: "others",
            },
          }}
        />
      );
      let component = wrapper.dive().dive().instance();
      expect(component.props.emtySelectOption).toBe(true);
      expect(component.props.multiple).toBe(true);
      const event = {
        target: { value: [] },
      };
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalled();
    });
    test("should call handleChange() has multiple and  has optionSelectAll  has emtySelectOption", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          emtySelectOption={true}
          multiple={true}
          value={["1"]}
          options={{
            emtyOption: {
              label: "-- Choose a option --",
              value: "choose-emty",
            },
            selectAll: {
              label: "Choose All",
              value: "choose-all",
            },
            lables: {
              others: "others",
            },
          }}
          optionSelectAll={true}
        />
      );
      let component = wrapper.dive().dive().instance();
      expect(component.props.emtySelectOption).toBe(true);
      expect(component.props.multiple).toBe(true);
      expect(component.props.optionSelectAll).toBe(true);
      const event = {
        target: { value: [] },
      };
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalled();
    });
    test("should call handleChange() select an option", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          emtySelectOption={true}
          multiple={true}
          value={[]}
          options={{
            emtyOption: {
              label: "-- Choose a option --",
              value: "choose-emty",
            },
            selectAll: {
              label: "Choose All",
              value: "choose-all",
            },
            lables: {
              others: "others",
            },
          }}
          optionSelectAll={true}
        />
      );
      let component = wrapper.dive().dive().instance();
      const event = {
        target: { value: ["Radio3"] },
      };
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalledWith(["Radio3"]);
    });
    test("should call handleChange() select all", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          emtySelectOption={true}
          multiple={true}
          value={[]}
          options={{
            emtyOption: {
              label: "-- Choose a option --",
              value: "choose-emty",
            },
            selectAll: {
              label: "Choose All",
              value: "choose-all",
            },
            lables: {
              others: "others",
            },
          }}
          optionSelectAll={true}
        />
      );
      let component = wrapper.dive().dive().instance();
      const event = {
        target: { value: ["Radio1", "Radio2", "Radio3", "Radio4"] },
      };
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalledWith([
        "Radio1",
        "Radio2",
        "Radio3",
        "Radio4",
      ]);
    });
    test("should call handleChange() Un select all", () => {
      const handleChange = jest.fn();
      const items: ISelectItem[] = [
        { label: "Radio1", value: "Radio1" },
        { label: "Radio2", value: "Radio2" },
        { label: "Radio3", value: "Radio3" },
        { label: "Radio4", value: "Radio4" },
      ];
      let wrapper = shallow(
        <Component
          selectItems={items}
          onChange={handleChange}
          id="DE"
          multiple={true}
          value={["Radio1", "Radio2", "Radio3", "Radio4"]}
          options={{
            emtyOption: {
              label: "-- Choose a option --",
              value: "choose-emty",
            },
            selectAll: {
              label: "Choose All",
              value: "choose-all",
            },
            lables: {
              others: "others",
            },
          }}
          optionSelectAll={true}
        />
      );
      let component = wrapper.dive().dive().instance();
      const event = {
        target: { value: ["Radio1", "Radio2", "Radio3"] },
      };
      component.state.values = ["choose-all", "Radio1", "Radio2", "Radio3"];
      component.state.seletedAll = true;
      component.handleChange(event);
      expect(handleChange).toHaveBeenCalledWith([]);
    });
  });
  test("should call handleFocus() Un select all", () => {
    const handleChange = jest.fn();
    const handleFocus = jest.fn();
    const handleBlur = jest.fn();
    const items: ISelectItem[] = [
      { label: "Radio1", value: "Radio1" },
      { label: "Radio2", value: "Radio2" },
      { label: "Radio3", value: "Radio3" },
      { label: "Radio4", value: "Radio4" },
    ];
    let wrapper = shallow(
      <Component
        selectItems={items}
        onChange={handleChange}
        onFocus={handleFocus}
        onBlur={handleBlur}
        id="DE"
        multiple={true}
        value={["Radio1", "Radio2", "Radio3", "Radio4"]}
        options={{
          emtyOption: {
            label: "-- Choose a option --",
            value: "choose-emty",
          },
          selectAll: {
            label: "Choose All",
            value: "choose-all",
          },
          lables: {
            others: "others",
          },
        }}
        optionSelectAll={true}
      />
    );
    let component = wrapper.dive().dive().instance();
    const event: React.ChangeEvent<{}> = {};
    component.handleFocus(event);
    component.handleBlur(event);
    expect(handleBlur).toHaveBeenCalledWith(event);
    expect(handleBlur).toHaveBeenCalledWith(event);
  });
  test("should call error ", () => {
    const handleChange = jest.fn();
    const items: ISelectItem[] = [
      { label: "Radio1", value: "Radio1" },
      { label: "Radio2", value: "Radio2" },
      { label: "Radio3", value: "Radio3" },
      { label: "Radio4", value: "Radio4" },
    ];
    const { container } = render(
      <Component
        selectItems={items}
        onChange={handleChange}
        id="DE"
        multiple={true}
        value={["Radio1", "Radio2", "Radio3", "Radio4"]}
        options={{
          emtyOption: {
            label: "-- Choose a option --",
            value: "choose-emty",
          },
          selectAll: {
            label: "Choose All",
            value: "choose-all",
          },
          lables: {
            others: "others",
          },
        }}
        optionSelectAll={true}
        error={true}
        helperText={`Has error appear!`}
      />
    );
    const text = container.querySelector(".data-test-FormHelperText");
    expect(text?.textContent).toBe(`Has error appear!`);
  });
  test("should call handleRenderValue item = [] ", () => {
    const handleChange = jest.fn();
    const items: ISelectItem[] = [
      { label: "Radio1", value: "Radio1" },
      { label: "Radio2", value: "Radio2" },
      { label: "Radio3", value: "Radio3" },
      { label: "Radio4", value: "Radio4" },
    ];
    const wrapper = shallow(
      <Component
        selectItems={items}
        onChange={handleChange}
        id="DE"
        multiple={true}
        value={[]}
        optionSelectAll={true}
        emtySelectOption={true}
        options={{
          emtyOption: {
            label: "-- Choose a option --",
            value: "choose-emty",
          },
          selectAll: {
            label: "Choose All",
            value: "choose-all",
          },
          lables: {
            others: "others",
          },
        }}
      />
    );
    const component = wrapper.dive().dive().instance();
    const data = component.handleRenderValue([]);
    expect(data).toBe(`-- Choose a option --`);
  });
  test("should call handleRenderValue item > 2", () => {
    const handleChange = jest.fn();
    const items: ISelectItem[] = [
      { label: "Radio1", value: "Radio1" },
      { label: "Radio2", value: "Radio2" },
      { label: "Radio3", value: "Radio3" },
      { label: "Radio4", value: "Radio4" },
    ];
    const wrapper = shallow(
      <Component
        selectItems={items}
        onChange={handleChange}
        id="DE"
        multiple={true}
        value={["Radio1", "Radio2", "Radio3"]}
        optionSelectAll={true}
        emtySelectOption={true}
        options={{
          emtyOption: {
            label: "-- Choose a option --",
            value: "choose-emty",
          },
          selectAll: {
            label: "Choose All",
            value: "choose-all",
          },
          lables: {
            others: "others",
          },
        }}
      />
    );
    const component = wrapper.dive().dive().instance();
    const data = component.handleRenderValue([
      "Radio1",
      "Radio2",
      "Radio3",
      "Radio4",
    ]);
    const lable = shallow(data);
    expect(lable.dive().text()).toBe(`Radio1, Radio2 ( + 1 others)`);
  });
  test("should call handleRenderValue item <= 2", () => {
    const handleChange = jest.fn();
    const items: ISelectItem[] = [
      { label: "Radio1", value: "Radio1" },
      { label: "Radio2", value: "Radio2" },
      { label: "Radio3", value: "Radio3" },
      { label: "Radio4", value: "Radio4" },
    ];
    const wrapper = shallow(
      <Component
        selectItems={items}
        onChange={handleChange}
        id="DE"
        multiple={true}
        value={["Radio1", "Radio2"]}
        optionSelectAll={true}
        emtySelectOption={true}
        options={{
          emtyOption: {
            label: "-- Choose a option --",
            value: "choose-emty",
          },
          selectAll: {
            label: "Choose All",
            value: "choose-all",
          },
          lables: {
            others: "others",
          },
        }}
      />
    );
    const component = wrapper.dive().dive().instance();
    const data = component.handleRenderValue(["Radio1", "Radio2"]);
    expect(data).toBe(`Radio1, Radio2`);
  });
  // test("should call handleBlur()", () => {
  //   const handleBlur = jest.fn();
  //   let wrapper = shallow(<Component onBlur={handleBlur} />);
  //   let component = wrapper.dive().dive().instance();
  //   component.handleBlur(null);
  //   expect(handleBlur).toHaveBeenCalled();
  // });
});
