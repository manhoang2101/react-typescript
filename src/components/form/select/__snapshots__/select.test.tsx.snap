// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSelect /> toMatchSnapshot() 1`] = `ShallowWrapper {}`;
